palantir-java-format/src/main/java/com/palantir/javaformat/java/Formatter.java

```diff
@@ -197,6 +211,24 @@ public CharSequence getCharContent(boolean ignoreEncodingErrors) throws IOExcept
         return unit;
     }
 
+    // Runtime.Version was added in JDK 9, so use reflection to access it to preserve source
+    // compatibility with Java 8.
+    @VisibleForTesting
+    static int getMajor() {
+        try {
+            Method versionMethod = Runtime.class.getMethod("version");
+            Object version = versionMethod.invoke(null);
+            return (int) version.getClass().getMethod("major").invoke(version);
+        } catch (Exception e) {
+            // continue below
+        }
+        int version = (int) Double.parseDouble(JAVA_CLASS_VERSION.value());
+        if (49 <= version && version <= 52) {
+            return version - (49 - 5);
+        }
+        throw new IllegalStateException("Unknown Java version: " + JAVA_SPECIFICATION_VERSION.value());
+    }
```

Comment:
this looked a bit bonkers to me, but I see it matches what they have in g-j-f!

https://github.com/google/google-java-format/blob/4ddb9148c103b9c65768e519884b0c8ad5caaf39/core/src/main/java/com/google/googlejavaformat/java/Formatter.java#L179-L194

Reply:
Actually hang on - we've specified our sourceCompat and targetCompat=11 now right? so we should be able to throw away all these reflective invocations, and just call Runtime.version() because that's available in java9 upwards???

Reply:
Yep, this is no longer necessary. Replaced with `Runtime.version().feature()` which got introduced in jdk9